tests/slsa_analyzer/test_git_url.py

```diff
@@ -122,36 +118,120 @@ def test_get_remote_vcs_url() -> None:
     assert git_url.get_remote_vcs_url("git@github.com:7999/org/") == ""
 
 
-def test_invalid_git_allowed_hosts() -> None:
+@pytest.mark.parametrize(
+    ("config_input", "expected_allowed_domain_set"),
+    [
+        (
+            """
+            [git_service.github]
+            domain = github.com
+
+            [git_service.gitlab.public]
+            domain = gitlab.com
+            """,
+            {"github.com", "gitlab.com"},
+        ),
+        (
+            """
+            [git_service.gitlab.public]
+            domain = gitlab.com
+
+            [git_service.gitlab.private]
+            domain = internal.gitlab.org
+            """,
+            {"gitlab.com", "internal.gitlab.org"},
+        ),
+    ],
+)
+def test_get_allowed_git_service_domains(
+    config_input: str,
+    expected_allowed_domain_set: set[str],
+) -> None:
+    """Test the get allowed git service domains function."""
+    config = configparser.ConfigParser()
+    config.read_string(config_input)
+    assert set(git_url.get_allowed_git_service_domains(config)) == expected_allowed_domain_set
+
+
+@pytest.mark.parametrize(
+    ("default_config_input", "override_config_input", "expected_allowed_domain_set"),
+    [
+        pytest.param(
+            # The current behavior is: we always enable GitHub and public GitLab by default.
+            # User config cannot disable either of the two.
+            """
+            [git_service.github]
+            domain = github.com
+
+            [git_service.gitlab.public]
+            domain = gitlab.com
+            """,
+            """
+            [git_service.github]
+            domain = github.com
+            """,
+            {"github.com", "gitlab.com"},
+            id="Only GitHub in user config",
+        ),
+        pytest.param(
+            """
+            [git_service.github]
+            domain = github.com
+
+            [git_service.gitlab.public]
+            domain = gitlab.com
+            """,
+            """
+            [git_service.gitlab.private]
+            domain = internal.gitlab.org
+            """,
+            {"github.com", "gitlab.com", "internal.gitlab.org"},
+            id="Private GitLab in user config",
+        ),
+    ],
+)
+def test_get_allowed_git_service_domains_with_override(
+    default_config_input: str,
+    override_config_input: str,
+    expected_allowed_domain_set: set[str],
+    tmp_path: Path,
```

Comment:
Where is `tmp_path` set? I couldn't find any fixture for it.

Reply:
It's a [pytest pre-defined fixture](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html). It creates temporary directories in `/tmp` that are unique to each invocation of `pytest`.

Reply:
Nice, I didn't know about it :slightly_smiling_face: